### ISSUE TITLE
Forwards cursor pagination works as expected, endCursor and startCursor gets swapped on backwards.

### ISSUE THREAD

**Issue by jonshipman** [wp-graphql/wp-graphql#1552]
#1411 might be the same issue, however, as I'm on 0.14 it could be an entirely different problem and fix.

Video illustrating the problem: https://youtu.be/ETBp_wXnIDQ

Forwards, all seems golden. However going backwards swaps the endCursor and startCursor causing different results to pull in.

Installed:
```
WordPress              5.5.3
wp-graphql-yoast-seo   4.8.0
advanced-custom-fields 5.9.3
custom-post-type-ui    1.8.1
wp-graphql             0.14.0
wp-graphql-custom-post-type-ui 1.1
wp-graphql-acf         0.3.4
wp-graphql-woocommerce 0.6.1
wordpress-seo          15.2.1
```

**Comment by ardiewen** [comment]
@jasonbahl Can we get this looked at? It's the same issue as the one I reported in #1411, and I can confirm it's still happening in WP GraphQL 1.0. For those of us using WP GraphQL without the experimental source plugin (or not using Gatsby at all), this prevents us from doing meaningful on-page pagination.

Please let me know if you'd like an email invite to a minimal repro wp-admin demonstrating the issue. This can be 100% replicated using the built in GraphiQL IDE and is frontend agnostic.

**Comment by ardiewen** [comment]
@jonshipman if you read through my original issue, my workaround currently is to invert the "order" (e.g. ASC to DESC) when paginating backwards. You will get your intended backwards pagination results this way.

However, be forewarned as this does not appear to be the intent of the Relay pagination spec described here: https://relay.dev/graphql/connections.htm

**Comment by etoler1841** [comment]
Running into this issue as well. When going flipping between an `after`-based query and a `before`-based query, as @ardiewen said, you need to flip the order of the sort. But once you get a response, if this is a `before` query, you also have to reverse the order of the results and switch the startCursor and endCursor. This works, but it is definitely not the intended implementation.

**Comment by IvanCoronado** [comment]
Still happening with v1.1.1 published 3 days ago

**Comment by hallee** [comment]
I'm seeing this too. Our solution was to swap the cursors in `graphql_connection_page_info`:

```php
/**
  * Swap the startCursor and endCursor for reverse pagination.
  * See: https://github.com/wp-graphql/wp-graphql/issues/1552
  */
 add_filter( 'graphql_connection_page_info', function( $page_info, $resolver ) {
 	if ( ! empty( $resolver->getArgs()['last'] ) ) {
 		$start_cursor = $page_info['startCursor'];
 		$page_info['startCursor'] = $page_info['endCursor'];
 		$page_info['endCursor'] = $start_cursor;
 	}
 	return $page_info;
 }, 10, 2 );
 ```

**Comment by jasonbahl** [comment]
👋  @jonshipman, @ardiewen, @etoler1841, @IvanCoronado, @hallee 

Sorry for the delay, but I believe this is fully resolved by #1773. You should no longer need to reverse cursors like @hallee is doing. 

I'm going to close this as I believe that it's resolved by #1773 and there's some pretty good tests in place to ensure this behavior continues to work as expected now. See: https://github.com/wp-graphql/wp-graphql/pull/1773/files#diff-e9fc42ba1f4b498640f9c212a0f768596ae74b0509003e669f23d7886c7bbb31R84-R587

If #1773 doesn't fully resolve this, please open a new issue with any added details and reference this one for context. 

Thanks ya'll!
